File: tests/fixtures/fizzbuzz.rb
# FizzBuzz implementation with a deliberate bug
# Bug: Line 9 uses % 4 instead of % 5 for "Buzz"

def fizzbuzz(n)
  if n % 15 == 0
    "FizzBuzz"
  elsif n % 3 == 0
    "Fizz"
  elsif n % 4 == 0  # BUG: should be % 5
    "Buzz"
  else
    n.to_s
  end
end

def main
  (1..100).each do |i|
    puts fizzbuzz(i)
  end
end

main if __FILE__ == $0

File: tests/fixtures/fizzbuzz.js
// FizzBuzz implementation with deliberate bug for debugging
// Bug: Line 9 uses n % 4 instead of n % 5 for "Buzz"

function fizzbuzz(n) {
    if (n % 15 === 0) {
        return "FizzBuzz";
    } else if (n % 3 === 0) {
        return "Fizz";
    } else if (n % 4 === 0) {  // BUG: Should be n % 5
        return "Buzz";
    } else {
        return n.toString();
    }
}

// Main execution
for (let i = 1; i <= 100; i++) {
    console.log(fizzbuzz(i));
}
